src/error.rs

```diff
@@ -0,0 +1,33 @@
+use defines::AfError;
+
+pub type ErrorCallback = Fn(AfError);
+
+pub static mut HANDLE_ERROR: &'static ErrorCallback = &handle_error_general; 
+
+pub fn register_error_handler(callback: &'static ErrorCallback) {
+    unsafe {
+        HANDLE_ERROR = callback;
+    }
+}
+
+pub fn handle_error_general(error_code: AfError) {
+    match error_code {
+        AfError::SUCCESS            => {}, /* No-op */
```

Comment:
Ah ok, that makes sense. In this case it would only need a mutex if you think people would be modifying the handler (at runtime) from multiple threads. If that is the case then yes it should be mutexed.